src/Lean/MetavarContext.lean

```diff
@@ -240,10 +240,12 @@ def LocalInstances.erase (localInsts : LocalInstances) (fvarId : FVarId) : Local
   | some idx => localInsts.eraseIdx idx
   | _        => localInsts
 
+/-- A kind for the metavariable that determines its unification behaviour.
+For more information see the large comment at the beginning of this file. -/
 inductive MetavarKind where
-  | natural
-  | synthetic
-  | syntheticOpaque
+  | /-- Normal unification behaviour -/ natural
```

Comment:
We are putting the docstrings before the `|` now.
See recent PR from @gebner

Reply:
It works in both places now, right? As a style point, I would suggest that we put it after the `|` if the comment is on the same line as the `|`, because otherwise the `|` will appear not column aligned which I think will be a bad idea following #1612. (We may or may not additionally recommend to *always* put docs on a separate line from the constructor name; it makes alignment in general kind of messy when you do this.)

Reply:
The column alignment will apply to the whole constructor node.  So if the docstring comes first, the docstring needs to be at the right column.  If the pipe comes first, then the pipe needs to be aligned.  Both styles will be possible.

Reply:
In general, I would rather tend to disallow docstring-after-the-pipe entirely (at elaboration time).